### PR TITLE
spinners: fix remove logic

### PR DESCRIPTION
**Pull request by benjamincburns**
Works around a bug in the `spinnies` module (jcarpanelli/spinnies#34) where removal of the final spinner may cause the spinner text to not disappear and/or may cause the process to hang on termination due to an interval not being cleared.

### CODE REVIEW

File: packages/events/defaultSubscribers/migrate/Messages.js [hunk 1]
```diff
@@ -220,7 +220,7 @@ class Messages {
       deployed: () => {
         let stopText;
         if (reporter.blockSpinner) {
-          reporter.blockSpinner.stop();
+          reporter.blockSpinner.remove();
```
Comment (cds-amal) [inline packages/events/defaultSubscribers/migrate/Messages.js:223]: Does `blockSpinner.remove()` have a meaningful implication to `stopText`? It also seems weird that, in the pre-commit `stop()` is invoked before setting `stopText`

**Review by cds-amal — CHANGES_REQUESTED** [review]
> Works around a bug in the spinnies module where removal of the final spinner may cause the spinner text to not disappear and/or may cause the process to hang on termination due to an interval not being cleared.

That's quite the premise! This should probably link to the bug/issue described.

File: packages/spinners/lib/spinner.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ export class Spinner {
       return;
     }
     spinnies.remove(this.name);
+    // spinnies.checkIfActiveSpinners();
```
[the remark this answers is not in the document]
Comment (benjamincburns) [reply packages/spinners/lib/spinner.ts:48]: Oof, I accidentally committed it like that while doing some manual testing to ensure that the change did in fact work around the issue.

File: packages/events/defaultSubscribers/migrate/Messages.js (hunk 1 above)
Comment (benjamincburns) [reply packages/events/defaultSubscribers/migrate/Messages.js:223]: `stopText` is unrelated to the spinner - I didn't write that code, but it appears to be used for logging output.

Comment (cds-amal) [reply packages/events/defaultSubscribers/migrate/Messages.js:223]: oh! I see. Thanks!

File: packages/spinners/typings/spinnies/index.d.ts [hunk 3]
```diff
@@ -131,5 +131,10 @@ declare module "spinnies" {
      * @returns false if all spinners have succeeded, failed or have been stopped
      */
     hasActiveSpinners: () => boolean;
+
+    /**
+     * Updates the spinning interval if no spinners are left active
+     */
+    checkIfActiveSpinners: () => void;
```
[the remark this answers is not in the document]
Comment (benjamincburns) [reply packages/spinners/typings/spinnies/index.d.ts:209]: Per other comment, this was necessary, as the invocation shouldn't have been commented out

**Comment by benjamincburns** [comment]
> This should probably link to the bug/issue described.

No such issue exists, and I didn't bother to create one as `spinnies` appears to be unmaintained. I'll go ahead and create one now, however.

I did reach out to the repo owner and offer to volunteer as a maintainer however, as it's a project that gets 100k+ weekly downloads and appears to see very few bug reports and/or pull requests.

**Comment by cds-amal** [comment]
> ... Issue created: [jcarpanelli/spinnies#34](https://github.com/jcarpanelli/spinnies/issues/34)
> 
> I did reach out to the repo owner and offer to volunteer as a maintainer however, as it's a project that gets 100k+ weekly downloads and appears to see very few bug reports and/or pull requests.

It's a better experience to ora, imo. If the author abandoned it, perhaps we can switch to a maintained fork, or fork our own.

**Pull request merged**